Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -616,6 +616,7 @@ void BackForwardCache::prune(PruningReason pruningReason)
 {
     while (pageCount() > maxSize()) {
         auto oldestItem = m_items.takeFirst();
+        auto cachedPage = m_cachedPageMap.take(oldestItem);
         m_cachedPageMap.set(oldestItem, pruningReason);
```

Comment:
Problem here is it _sure looks_ like you are accomplishing nothing, because removing an item from the map and then setting an item with the same key seems equivalent to just setting the item. Your commit message explains this is not so, but whoever modifies this code next doesn't know that. I know we generally try to avoid adding too many comments into WebKit code, but a tricky case like this really merits one. And although the tests you added are nice, they notably are not designed to catch a regression here.